src/main/scala/com/dwolla/cloudflare/DnsRecordClient.scala

```diff
@@ -36,9 +36,11 @@ class DnsRecordClient(executor: CloudflareApiExecutor)(implicit val ec: Executio
     }.map((_, record.zoneId))
   }
 
-  def getExistingDnsRecord(name: String): Future[Option[IdentifiedDnsRecord]] = {
+  def getExistingDnsRecord(name: String, content: Option[String]=None, recordType: Option[String]=None): Future[Option[IdentifiedDnsRecord]] = {
     getZoneId(domainNameToZoneName(name)).flatMap { zoneId ⇒
-      val request: HttpGet = new HttpGet(s"https://api.cloudflare.com/client/v4/zones/$zoneId/dns_records?name=$name")
+      val contentParameter = content.fold("")(c ⇒ s"?content=$c")
```

Comment:
I would do something like 

```scala
val parameters = List(Option("name" → name), content.map("content" → _), recordTypeParameter.map("type" → _))
  .collect {
    case Some((key, value)) ⇒ s"$key=$value"
  }
  .mkString("&")
```

and then replace `name=$name$contentParameter$recordTypeParameter` below with `$parameters`